src/mono/mono/metadata/assembly-internals.h

```diff
@@ -141,6 +141,30 @@ mono_assembly_get_image_internal (MonoAssembly *assembly);
 void
 mono_set_assemblies_path_direct (char **path);
 
+typedef enum {
+	BUNDLED_BLOB,
+	BUNDLED_ASSEMBLY,
+	BUNDLED_SATELLITE_ASSEMBLY,
+	BUNDLED_BINARY,
+	BUNDLED_RESOURCE_COUNT,
+} BundledResourceType;
+
+typedef struct BundledResource {
+	const char *culture; // Satellite assemblies
+	const unsigned char *data;
+	unsigned int size;
+	BundledResourceType type;
+} BundledResource;
+
+void
+mono_add_bundled_resource (const char *name, const char *culture, const unsigned char *data, unsigned int size, BundledResourceType type);
```

Comment:
These API's are not thread safe and I assume they must all be called before we init the runtime, maybe we should enforce that is the case and assert if called incorrectly.

Reply:
How do we enforce that this is only called before runtime is initialized? 
This case only concerns those that do not use the default runtime initialization callback right? Since we wouldn't know about what logic they use, we'd have to rely on mono APIs to see if the runtime was initialized?

We wouldn't be able to "try to init runtime and expect fail" as a check cause it might initialize the runtime right?
```
MonoDomain *domain = mono_jit_init ("should.not.be.initialized");
assert (!domain);
```